compiler/codegen/OMRRegisterIterator.cpp

```diff
@@ -19,47 +19,28 @@
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
  *******************************************************************************/
 
-#include <stddef.h>
-#include "codegen/Machine.hpp"
-#include "codegen/RealRegister.hpp"
-#include "codegen/Register.hpp"
-#include "codegen/RegisterIterator.hpp"
-#include "infra/Assert.hpp"
+#include <stddef.h>                        // for NULL
+#include "codegen/Machine.hpp"             // for Machine
+#include "codegen/RealRegister.hpp"        // for RealRegister, etc
+#include "codegen/Register.hpp"            // for Register
+#include "codegen/OMRRegisterIterator.hpp" // for RegisterIterator
+#include "infra/Assert.hpp"                // for TR_ASSERT
```

Comment:
I never really found these comments particularly useful and they have a high maintenance with code moving around. How do other committers feel about not including them?

Reply:
I'm don't mind removing the comments, no problem. @Leonardo2718 @0xdaryl @0dvictor What is your opinion about comments after includes, do we need them there?

Reply:
I'm okay with removing them. I've never found these kinds of comments useful either.

Reply:
I've removed the comments from the files touched by PR.

Reply:
Those `#include` comments are remnants from when we ran Include-What-You-Use over the code base.  They are impossible to maintain and keep up-to-date manually, and really only suitable if we regularly run IWYU on our code (which we don't).  I'm in favour of removing them.